pages/api/dpu/trigger.ts

```diff
@@ -11,13 +12,17 @@ const triggerHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 	if (!jsonBody.selectedProvider || !jsonBody.selectedInstallationType || !jsonBody.selectedHosting || !jsonBody.userId) {
 		console.error("[triggerHandler] Invalid request body");
 		res.status(400).json({ "error": "Invalid request body" });
+		const eventProperties = { ...jsonBody, response_status: 400 };
+		rudderStackEvents.track("", "", 'trigger', { type: 'run-trigger', eventStatusFlag: 0, eventProperties });
 		return;
 	}
-	
+
 	if (jsonBody.selectedProvider === 'github' && jsonBody.selectedInstallationType === 'individual' && jsonBody.selectedHosting === 'cloud') {
-		if(!jsonBody.github_pat){
+		if (!jsonBody.github_pat) {
 			console.error("[triggerHandler] Missing GitHub Personal Access Token");
 			res.status(400).json({ "error": "Missing GitHub Personal Access Token" });
+			const eventProperties = { ...jsonBody, response_status: 400 };
+			rudderStackEvents.track("", "", 'trigger', { type: 'run-trigger', eventStatusFlag: 0, eventProperties });
```

Comment:
We have the `userId` here, as ensured by the previous `if` block.

```suggestion
			rudderStackEvents.track(jsonBody.userId, "", 'trigger', { type: 'run-trigger', eventStatusFlag: 0, eventProperties });
```